js/src/ui/Form/AddressSelect/addressSelect.js

```diff
@@ -15,262 +15,799 @@
 // along with Parity.  If not, see <http://www.gnu.org/licenses/>.
 
 import React, { Component, PropTypes } from 'react';
-import { MenuItem } from 'material-ui';
-import { isEqual, pick } from 'lodash';
-
-import AutoComplete from '../AutoComplete';
-import IdentityIcon from '../../IdentityIcon';
-import IdentityName from '../../IdentityName';
-
+import ReactDOM from 'react-dom';
+import { connect } from 'react-redux';
+import Portal from 'react-portal';
+import keycode, { codes } from 'keycode';
+
+import CloseIcon from 'material-ui/svg-icons/navigation/close';
+import TextFieldUnderline from 'material-ui/TextField/TextFieldUnderline';
+
+import ParityBackground from '~/ui/ParityBackground';
+import IdentityIcon from '~/ui/IdentityIcon';
+import InputAddress from '~/ui/Form/InputAddress';
+import Tags from '~/ui/Tags';
 import { fromWei } from '~/api/util/wei';
 
 import styles from './addressSelect.css';
 
-export default class AddressSelect extends Component {
+class AddressSelect extends Component {
   static contextTypes = {
-    api: PropTypes.object.isRequired
-  }
+    muiTheme: PropTypes.object.isRequired
+  };
 
   static propTypes = {
+    // Required props
```

Comment:
General comment - as soon as we have to start explaining how we thing and ordered the props, it is not self-explanatory and easy to follow.